readthedocs/proxito/views/utils.py

```diff
@@ -85,4 +85,5 @@ def _get_project_data_from_request(
     # * Subproject
     # * Translations
 
+    filename = filename.rstrip('/')
```

Comment:
This seems hacky -- why are we doing this here?

Reply:
the slash gets captured in https://github.com/readthedocs/readthedocs.org/blob/dd66c224052d90cb08b3b7828ab0de205e069c0b/readthedocs/proxito/urls.py#L94-L94

and it can end up with a slash in https://github.com/readthedocs/readthedocs.org/blob/fix-trailing-slash/readthedocs/proxito/views/utils.py#L61-L61

Reply:
We should attempt to fix that closer to where it's happening then. Perhaps `os.path.join` was the proper logic after all?

 We are calling this function a lot of places, and I don't want to put the logic here because it might cause other unintended issues.

Reply:
os.path.join depends on the running OS to add `/` or `\`, it could also happen to add an extra `/` if it thinks it's a directory or the root. Changed to just strip the file before concatenate it.

Reply:
We have been hitting these kind of issues a lot when manipulating URLs, do you think that using a 3rd party library and follow the same pattern everywhere (*) in our code could help us?

3rd party package example: https://github.com/homm/yurl/

(*) do not use `os.path.join`, nor `"...".format()` nor `"..." + "..."` etc, that we have mixtures of all of them --and use a package like this intead

Reply:
Looks like doesn't work like we want

```
>>> print(yurl.URL('foo/bar') + yurl.URL('bar'))
foo/bar
>>> print(yurl.URL('foo') + yurl.URL('bar'))
bar
>>> print(yurl.URL('foo/') + yurl.URL('bar'))
foo/bar
>>> print(yurl.URL('foo/') + yurl.URL('/bar'))
/bar
```

I think we could just write a little function to handle it.

Reply:
Yeah. I haven't tried that particular one, but there are many similar to this one.